### PR TITLE
 Improve websocket's flow (avoid duplicate messages / sub&unsub)

### PR DESCRIPTION
**Pull request by Johnny850807**
Fix: #46 

 Improve websocket's flow (avoid duplicate messages / sub&unsub)

### CODE REVIEW

File: src/app/services/Services.ts [hunk 1]
```diff
@@ -86,10 +86,13 @@ export abstract class StudentService {
...
 
   set currentStudent(student: Student) {
+    const newStudent = this._currentStudent?.id !== student?.id;
     this._currentStudent = student;
     if (student) {
       console.log(`Set current student to ${studentToString(student)}.`);
-      this.student$.next(student);
+      if (newStudent) {
+        this.student$.next(student);
+      }
```
Comment (Johnny850807) [inline src/app/services/Services.ts:95]: Only broadcast on the new student login.

File: src/app/services/impl/StompBrokerService.ts [hunk 2]
```diff
@@ -21,30 +22,48 @@ export class StompBrokerService extends BrokerService {
     if (!this.stompClient.active) {
       this.stompClient.configure(this.stompConfig);
       this.stompClient.activate();
-      this.connect$.next();
+      this.stompClient.connectionState$.subscribe((state: RxStompState) => {
+        console.log(`RxStomp's state: ${RxStompState[state]}`);
```
Comment (edisonhello) [inline src/app/services/impl/StompBrokerService.ts:26]: Is this a debug message or a useful message you want to keep in console?

**Review by edisonhello — APPROVED** [review]
LGTM.

File: src/app/services/impl/StompBrokerService.ts (hunk 2 above)
Comment (Johnny850807) [reply src/app/services/impl/StompBrokerService.ts:26]: We would like to keep this since we are going to troubleshoot over a ton of WebSockets stuff

**Pull request merged**